api/httpapi/handlers/asset_handler_test.go

```diff
@@ -676,37 +676,60 @@ func TestAssetHandlerGet(t *testing.T) {
 			Description:  `should return http 500 if fetching fails`,
 			ExpectStatus: http.StatusInternalServerError,
 			Setup: func(ctx context.Context, ar *mocks.AssetRepository) {
-				ar.On("GetAll", ctx, asset.Config{}).Return([]asset.Asset{}, errors.New("unknown error"))
+				ar.On("GetAll", ctx, asset.Config{
+					Data: make(map[string]string),
+				}).Return([]asset.Asset{}, errors.New("unknown error"))
 			},
 		},
 		{
 			Description:  `should return http 500 if fetching total fails`,
 			Querystring:  "?with_total=1",
 			ExpectStatus: http.StatusInternalServerError,
 			Setup: func(ctx context.Context, ar *mocks.AssetRepository) {
-				ar.On("GetAll", ctx, asset.Config{}).Return([]asset.Asset{}, nil, nil)
+				ar.On("GetAll", ctx, asset.Config{
+					Data: make(map[string]string),
+				}).Return([]asset.Asset{}, nil, nil)
 				ar.On("GetCount", ctx, asset.Config{}).Return(0, errors.New("unknown error"))
 			},
 		},
 		{
 			Description:  `should parse querystring to get config`,
-			Querystring:  "?text=asd&type=table&service=bigquery&size=30&offset=50",
+			Querystring:  "?types=table&services=bigquery&size=30&offset=50&sort=created_at&direction=desc&config.data[dataset]=booking&config.data[project]=p-godata-id",
```

Comment:
need help adding more test cases for:
1. `q` and `q_fields` querystring
2. `data` querystring

You can create new test cases if you want instead of piling them all here.